File: SpotifyAPI.DAL/Models/PlaylistModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpotifyAPI.DAL.Models
{
    public class PlaylistModel : ErrorModel
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public bool? IsPublic { get; set; }
        public int? UserID { get; set; }
        public UserModel?  Owner { get; set; }
        public List<ImageModel>? Images { get; set; } 
        public string? Image { get; set; }
        public int? SongsCount { get; set; }
        public int? Duration { get; set; }

    }
    public class PlaylistException : Exception
    {
        public PlaylistException(string message) : base(message) { }
    }
}

File: SpotifyAngular/src/app/Models/playListModel.ts
import { ImageModel } from "./imageModel"
import { UserModel } from "./userModel"

export type PlaylistModel={
    id:number,
    name:string,
    description:string,
    isPublic?:boolean,
    userID:number,
    owner?:UserModel,
    images?:ImageModel[],
    songsCount?:number,
    duration?:number
}

export type PlayListDataSource={
    position:number,
    id:number,
    name:string,
    description:string,
    songsCount:number,
    duration:number,
    image:string,
    userID:number,
    isPublic:boolean
}